lib/src/Test/Pos/Helpers.hs

```diff
@@ -34,27 +39,52 @@ canonicalJsonTest ::
 canonicalJsonTest =
     identityTest @a $ \x ->
         canonicalJsonRenderAndDecode x .&&. canonicalJsonPrettyAndDecode x
+
+-- | Basically the same as `canonicalJsonTest` but tests a given `Gen a`.
```

Comment:
If we are sure of this, we could redefine:

```
canonicalJsonTest = canonicalJsonTest' arbitrary
```

But probably not worth the bug-introduction risk.

Reply:
@mhuesch That's what I was most worried about. Will give it some thought in the morning.

Reply:
Let's leave it as-is.